jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/DeploymentScanner.java

```diff
@@ -978,6 +981,13 @@ else if (FileID.isExtension(file, "xml"))
             }
         }
 
+        // Add deprecation warnings
+        envLayer.getAttributeNameSet().stream()
+            .filter((name) -> name.startsWith("jetty.deploy.environmentXml.") ||
+                name.equals("jetty.deploy.environmentXml"))
+            .forEach((name) ->
+                LOG.warn("Deprecated attribute key prefix detected {} (use ${jetty.base}/environments/*.xml instead)", name));
+
```

Comment:
why not rename `stripOldAttributePrefex` to `checkAttributeName` and move these tests into it?

Reply:
Fixed, but in a different way.